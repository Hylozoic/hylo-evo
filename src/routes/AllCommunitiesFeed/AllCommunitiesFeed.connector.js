import { connect } from 'react-redux'
import { getMe } from 'store/selectors/getMe'
import getParam from 'store/selectors/getParam'
import getQueryParam from 'store/selectors/getQueryParam'
import { FETCH_POSTS } from 'store/constants'
import { mapDispatchToProps } from 'routes/CommunityFeed'

export function mapStateToProps (state, props) {
  const filter = getQueryParam('t', state, props)
  const sortBy = getQueryParam('s', state, props)

  return {
    filter,
    sortBy,
    selectedPostId: getParam('postId', state, props),
    pending: state.pending[FETCH_POSTS],
    currentUser: getMe(state, props)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
