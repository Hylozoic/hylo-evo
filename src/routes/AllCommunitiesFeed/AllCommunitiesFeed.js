import React from 'react'
import './AllCommunitiesFeed.scss'
import Feed from 'components/Feed'
import FeedBanner from 'components/FeedBanner'
import { pick } from 'lodash/fp'
import { ALL_COMMUNITIES_ID } from 'components/Feed/Feed.store'

export default function AllCommunitiesFeed (props) {
  const { currentUser } = props
  const feedProps = {
    subject: 'all-communities',
    id: ALL_COMMUNITIES_ID,
    showCommunities: true,
    ...pick([
      'filter',
      'sortBy',
      'changeSort',
      'changeTab',
      'showPostDetails',
      'selectedPostId'
    ], props)
  }

  return <div styleName='container'>
    <FeedBanner all currentUser={currentUser} />
    <Feed {...feedProps} />
  </div>
}
