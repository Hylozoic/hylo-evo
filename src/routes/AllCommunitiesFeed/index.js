import AllCommunitiesFeed from './AllCommunitiesFeed'
import connector from './AllCommunitiesFeed.connector'
export default connector(AllCommunitiesFeed)
