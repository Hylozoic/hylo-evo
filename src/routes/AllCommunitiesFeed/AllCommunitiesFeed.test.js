import AllCommunitiesFeed from './AllCommunitiesFeed'
import { shallow } from 'enzyme'
import React from 'react'

it('renders as expected', () => {
  const wrapper = shallow(<AllCommunitiesFeed filter='request' />)
  expect(wrapper).toMatchSnapshot()
})
