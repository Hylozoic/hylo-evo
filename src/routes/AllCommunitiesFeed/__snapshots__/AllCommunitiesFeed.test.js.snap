// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  data-styleName="container"
>
  <FeedBanner
    all={true}
  />
  <Connect(Feed)
    filter="request"
    id="all-communities"
    showCommunities={true}
    subject="all-communities"
  />
</div>
`;
